pyvista/core/filters/poly_data.py

```diff
@@ -2756,6 +2766,11 @@ def extrude_rotate(
         alg.SetDeltaRadius(dradius)
         alg.SetCapping(capping)
         alg.SetAngle(angle)
+        if pyvista.vtk_version_info >= (9, 1, 0):
+            alg.SetRotationAxis(rotation_axis)
+        elif rotation_axis != (0, 0, 1):  # pragma: no cover
+            warnings.warn("rotation_axis is not supported with vtk < 9.1.0. Z-axis is used.")
```

Comment:
I think we should raise here instead as the behavior will be wrong compared to the user intention.

Reply:
```suggestion
            raise VTKVersionError('vtkRotationalExtrusionFilter.SetRotationAxis requires vtk>=9.1.0')
```